packages/alfa-css/src/grammars/color.ts

```diff
@@ -101,6 +109,121 @@ function rgbaColor(stream: Stream<Token>): Color {
   return color;
 }
 
+function hslaColor(stream: Stream<Token>): Color {
+  const args = functionArguments(stream);
+
+  if (args.length !== 3 && args.length !== 4) {
+    return Transparent;
+  }
+
+  let h = 0;
+  let s = 0;
+  let l = 0;
+  let a = 1;
+
+  for (let i = 0, n = args.length; i < n; i++) {
+    const component = args[i];
+
+    switch (i) {
+      case 0:
+      case 3:
+        if (args[i].type !== TokenType.Number) {
+          return Transparent;
+        }
+        break;
+      case 1:
+      case 2:
+        if (args[i].type !== TokenType.Percentage) {
+          return Transparent;
+        }
+    }
+
+    let { value } = <Number | Percentage>component;
+
+    if (i === Component.Alpha) {
+      value = clamp(value, 0, 1);
+    } else {
+      if (component.type === TokenType.Percentage) {
+        value *= 0x64;
+      }
+
+      value = clamp(value, 0, 360);
+    }
+
+    switch (i) {
+      case Component.Red:
+        h = value;
+        break;
+      case Component.Green:
+        s = value;
+        break;
+      case Component.Blue:
+        l = value;
+        break;
+      case Component.Alpha:
+        a = value;
+    }
+  }
+
+  const { r, g, b } = hslToRgb(h / 360, s / 100, l / 100);
+
+  return { red: r, green: g, blue: b, alpha: a };
+}
+
+/**
+ * Converts an HSL color value to RGB. Conversion formula
+ * adapted from http://en.wikipedia.org/wiki/HSL_color_space.
+ * Assumes h, s, and l are contained in the set [0, 1] and
+ * returns r, g, and b in the set [0, 255].
+ *
+ * @see https://stackoverflow.com/questions/2353211/hsl-to-rgb-color-conversion
```

Comment:
The function will have to be reimplemented; the code below is not licensed. Use: https://www.w3.org/TR/css-color/#hsl-to-rgb

Reply:
...and add a copyright notice and a license link to https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document.